src/compiler/checker.ts

```diff
@@ -24830,7 +24843,7 @@ namespace ts {
                     }
                     results.push(propType);
                 }
-                return getIntersectionType(results);
+                return getIntersectionType(results); // Same result for both union and intersection signatures

```

Comment:
The _actual_ surprising thing _is_ that nothing changed here. This code path handled union composite signatures... and now we're handling intersection composite signatures _in the same way_ (rather than inverting anything), which is _odd_.  The reason for that is explained in this comment (namely that for compat reasons we intentionally do "the wrong thing" for union signatures here, which happens to be "the right thing" for intersection signatures)